platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ZipUtils.java

```diff
@@ -0,0 +1,284 @@
+   * @throws MigrationException when the key can not be written to a file
+   * @return a {@link SecretKey}
+   */
+  public static SecretKey createSecretKey(Path keyPath) throws NoSuchAlgorithmException {
+    KeyGenerator keyGenerator = null;
+    try {
+      keyGenerator = KeyGenerator.getInstance(KEY_ALGORITHM);
+      SecretKey secretKey = keyGenerator.generateKey();
+      char[] hexKey = encodeHex(secretKey.getEncoded());
+      writeStringToFile(keyPath.toFile(), String.valueOf(hexKey), Charsets.UTF_8);
+      return secretKey;
+    } catch (NoSuchAlgorithmException e) {
+      LOGGER.error(EXPORT_KEY_ALGORITHM_ERROR, e);
+      throw e;
+    } catch (IOException e) {
+      throw new MigrationException(Messages.EXPORT_KEY_IO_ERROR, keyGenerator, e);
+    }
+  }
+
+  /**
+   * Loads an encryption key from a file
+   *
+   * @param keyPath the path to the key file
+   * @throws MigrationException when the key file can not be opened
+   * @throws MigrationException when the key stored in the file cannot be decoded
+   * @return the {@link SecretKey} loaded from the key file
+   */
+  @SuppressWarnings("squid:S2095" /* stream is closed in the finally clause */)
+  public static SecretKey loadDecryptionKeyFrom(Path keyPath) {
+    BufferedInputStream bif = null;
+    try {
+      bif = new BufferedInputStream(new FileInputStream(keyPath.toFile()));
+      String keyData = IOUtils.toString(bif, Charsets.UTF_8);
+      byte[] keyBytes = decodeHex(keyData.toCharArray());
+      return new SecretKeySpec(keyBytes, KEY_ALGORITHM);
+    } catch (IOException e) {
+      throw new MigrationException(Messages.IMPORT_KEY_OPEN_ERROR, keyPath, e);
+    } catch (DecoderException e) {
+      throw new MigrationException(Messages.IMPORT_KEY_DECODE_ERROR, keyPath, e);
+    } finally {
+      IOUtils.closeQuietly(bif);
+    }
+  }
+
+  /**
+   * Creates a {@link Cipher} for a key stored in a file
+   *
+   * @param keyPath the path to the key file
+   * @throws MigrationException when the key algorithm is invalid
+   * @throws MigrationException when the key padding is invalid
+   * @throws MigrationException when the key is invalid
+   * @return the {@link Cipher} generated from a key
+   */
+  public static Cipher loadDecryptionCipherFor(Path keyPath)
+      throws NoSuchAlgorithmException, NoSuchPaddingException {
+    try {
+      Cipher cipher = Cipher.getInstance(CIPHER_ALGORITHM);
+      cipher.init(Cipher.DECRYPT_MODE, loadDecryptionKeyFrom(keyPath));
+      return cipher;
+    } catch (NoSuchAlgorithmException e) {
+      LOGGER.error(IMPORT_KEY_ALGORITHM_ERROR, keyPath, e);
+      throw e;
+    } catch (NoSuchPaddingException e) {
+      LOGGER.error(IMPORT_KEY_PADDING_ERROR, keyPath, e);
+      throw e;
+    } catch (InvalidKeyException e) {
+      throw new MigrationException(Messages.IMPORT_KEY_INVALID_ERROR, keyPath, e);
+    }
+  }
+
+  public static String getChecksum(Path path) throws IOException, NoSuchAlgorithmException {
```

Comment:
❗️ Can we not use PathUtils.getChecksumFor()?

Reply:
I didn't even realize you already had that in there. I can change it

Reply:
Less code to maintain and test ;-)

Reply:
I'm actually going to keep this I think @paouelle , since the PathUtils stuff is all non-static and also uses a different hash algorithm for the checksum method